src/commands.rs

```diff
@@ -4022,78 +4050,138 @@ fn cmd_branch(ui: &mut Ui, command: &CommandHelper, args: &BranchArgs) -> Result
                     branch_name
                 ))),
             })
-            .try_collect()?
-    } else {
-        args.names.iter().map(|name| name.as_str()).collect()
-    };
+            .try_collect()?;
+        Ok(result)
+    }
 
-    if branch_names.is_empty() {
-        ui.write_warn("warning: No branches provided.\n")?;
+    fn make_branch_term(ui: &mut Ui, branch_names: &[impl AsRef<str>]) -> io::Result<String> {
+        if branch_names.is_empty() {
+            ui.write_warn("warning: No branches provided.\n")?;
```

Comment:
This can possibly be removed if the `#[clap(required = true)]` thing I suggested above handles all the cases